src/scripts/baseball/background/BgBaseball.gd

```diff
@@ -0,0 +1,61 @@
+class_name BgBaseball
+extends KinematicBody2D
+
+export var BASEBALL_GRAVITY = 0.0 
+
+export var MIN_BASEBALL_SCALE = 0.1
+export var MAX_BASEBALL_SCALE = 0.6
+
+export var BASEBALL_SCALE_STEP = 0.15
+export var BASEBALL_VELOCITY_SCALER_STEP = 1.2;
+ 
+#export var MAX_DISTANCE = 300 
+export var BOTTOM_DESTROY_BOUNDS = Globals.SCREEN_HEIGHT + 10
+export var TOP_DESTROY_BOUNDS = -10
+export var RIGHT_DESTROY_BOUNDS = Globals.SCREEN_WIDTH + 10
+export var LEFT_DESTROY_BOUNDS = -10
+
+
+onready var sprite = $Sprite
+onready var collision_shape = $CollisionShape2D
+onready var timer = $Timer
+var current_velocity: Vector2
+var ball_origin:Vector2
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	sprite.scale = Vector2(MIN_BASEBALL_SCALE, MIN_BASEBALL_SCALE)
+	timer.start()
+	timer.connect("timeout", self, "on_timeout")
+	ball_origin = global_position
+	pass # Replace with function body.
+	
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+#func _process(delta):
+#	pass
+
+func _physics_process(_delta):
+		move_and_collide(current_velocity , false, true, false)
+		if (global_position.y > BOTTOM_DESTROY_BOUNDS || 
```

Comment:
I think we should handle this with Godot's collision system so we can keep collision logic consistent across our entities. we can create new Area2Ds and put them on a collision layer specifically for baseball destroy bounds. that way we can hook into native signals from collision bodies, such as using Area2D nodes on the baseball to detect collisions and their area_entered or area_exited signals to handle these collisions. if there's any help needed here with setting up the masks or layers let me know!

always open to other opinions though!!

Reply:
My only concern is that this seems like it may introduce unneeded complexity with managing an extra collision layer / mask & bounding box since the bgBaseball isn't interacting with anything outside of the screen edge.

But if its the way godot should be written, than I shall write. 

![cat-typing-gif](https://user-images.githubusercontent.com/6611689/232839400-4801a4dd-15c6-4b91-b124-577afc45387b.gif)

Reply:
yeah that's a fair point! I don't think it's much complexity to have the collision layers for this since it's all built into Godot physics system, but I can't really foresee any current issues with your implementation. if you're feeling like you want to leave it as is that's fine with me as well!